compliance/src/main/java/bio/terra/pearl/compliance/SyncVantaUsers.java

```diff
@@ -208,25 +211,17 @@ private <T extends VantaObject> Collection<T> collectAllData(WebClient wc, Param
         AtomicInteger cursorCount = new AtomicInteger(0);
 
         while (cursorCount.get() == 0 || results.get().getPageInfo().isHasNextPage()) {
-            log.debug("Getting cursor number {}", cursorCount);
+            log.info("Getting {} cursor number {}", type.getType().getTypeName(), cursorCount);
 
-            add429StatusHander(wc.get().uri(uriBuilder -> {
-                if (cursorCount.get() > 0) {
-                    uriBuilder.queryParam("pageCursor", results.get().getPageInfo().getEndCursor());
-                }
-                uriBuilder.queryParam("pageSize", 100);
-                return uriBuilder.build();
-            }).retrieve());
-            VantaResultsResponse<T> response = add429StatusHander(wc.get().uri(uriBuilder -> {
+            VantaResultsResponse<T> response = (wc.get().uri(uriBuilder -> {
                         if (cursorCount.get() > 0) {
                             uriBuilder.queryParam("pageCursor", results.get().getPageInfo().getEndCursor());
                         }
                         uriBuilder.queryParam("pageSize", 100);
                         return uriBuilder.build();
-                    }).retrieve())
+                    }).retrieve()).onStatus(HttpStatus.TOO_MANY_REQUESTS::equals, get429StatusHander())
                     .bodyToMono(type)
-                    .retryWhen(Retry.withThrowable(throwableFlux -> throwableFlux.filter(RateLimitException.class::isInstance).map(t ->
```

Comment:
Recently vanta made a breaking change to one of their APIs, which exposed a bug in our code that led to the job hanging because it couldn't respond to a 400.  This bug is now fixed.